src/dayview.html

```diff
@@ -1,65 +1,167 @@
+          [initPosition]="initScrollPosition">
+          <table class="table table-bordered table-fixed dayview-normal-event-table">
+            <tbody>
+              <tr *ngFor="let tm of views[sliderIndex].rows; let i = index">
+                <td class="calendar-hour-column text-center">
+                  {{hourColumnLabels[i]}}
+                </td>
+                <td class="calendar-cell">
+                  <ng-template [ngTemplateOutlet]="dayviewInactiveNormalEventSectionTemplate"
+                    [ngTemplateOutletContext]="{tm:tm, hourParts: hourParts}">
+                  </ng-template>
+                </td>
+              </tr>
+            </tbody>
+          </table>
+        </init-position-scroll>
+      </ng-container>
+
+      <ng-container *ngIf="dayviewShowCategoryView">
+        <div *ngIf="sliderIndex===currentViewIndex" class="dayview-category-header">
+          <table class="table table-fixed dayview-category-header-table">
+            <thead>
+              <tr>
+                <th class="dayview-category-header-item calendar-hour-column"></th>
+                <th *ngFor="let category of views[sliderIndex].categories" class="dayview-category-header-item">
+                  <div class="dayview-category-name">
+                    <ng-template [ngTemplateOutlet]="dayviewCategoryItemTemplate"
+                      [ngTemplateOutletContext]="{category}">
+                    </ng-template>
+                  </div>
+                </th>
+              </tr>
+            </thead>
+          </table>
+        </div>
+        <div *ngIf="sliderIndex!==currentViewIndex" class="dayview-inactive-category-header"></div>
+
+        <div class="dayview-allday-table">
+          <div class="dayview-allday-label">{{ allDayLabel }}</div>
+          <div class="dayview-allday-content-wrapper scroll-content">
+            <table class="table table-fixed dayview-allday-category-table">
+              <tbody>
+                <td class="dayview-allday-category-cell" *ngFor="let category of views[sliderIndex].categories">
+                  <table class="table dayview-allday-content-table">
                     <tbody>
-                        <tr *ngFor="let tm of views[0].rows; let i = index">
-                            <td class="calendar-hour-column text-center">
-                                {{hourColumnLabels[i]}}
-                            </td>
-                            <td class="calendar-cell">
-                                <ng-template [ngTemplateOutlet]="dayviewInactiveNormalEventSectionTemplate"
-                                    [ngTemplateOutletContext]="{tm:tm, hourParts: hourParts}">
-                                </ng-template>
+                      <tr>
+                        <ng-container *ngIf="sliderIndex === currentViewIndex">
+                          <ng-container *ngFor="let allDayEventsCategory of views[sliderIndex].categorizedAllDayEvents">
```

Comment:
use categorizedAllDayEventsMap, then you don't need this for loop, simply fetch category as the key, in td directly use views[sliderIndex].categorizedAllDayEventsMap[category]